### PR TITLE
Add rule S6478 (react/no-unstable-nested-components): React components should not be nested

### LINKED ISSUE #3576: Add rule `react/no-unstable-nested-components`
RSPEC https://github.com/SonarSource/rspec/pull/1413

### PR DESCRIPTION
**Pull request by francoismora**
Fixes #3576

### CODE REVIEW

File: eslint-bridge/src/linting/eslint/rules/decorators/no-unstable-nested-components.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { Rule } from 'eslint';
+import { interceptReport } from './helpers';
+
+export function decorateNoUnstableNestedComponents(rule: Rule.RuleModule): Rule.RuleModule {
+  return interceptReport(rule, messageChanger(urlRemover()), contextOverriderForReact);
+}
+
+// interceptReport() by default doesn't work with the React plugin
+// as the rules fail to find the context getFirstTokens() function.
+function contextOverriderForReact(
```
Comment (ilia-kebets-sonarsource) [inline eslint-bridge/src/linting/eslint/rules/decorators/no-unstable-nested-components.ts:31]: What do you think about extracting this in helpers?

Comment (francoismora) [reply eslint-bridge/src/linting/eslint/rules/decorators/no-unstable-nested-components.ts:31]: Yes indeed that makes sense.

Comment (ilia-kebets-sonarsource) [reply eslint-bridge/src/linting/eslint/rules/decorators/no-unstable-nested-components.ts:31]: maybe event move the extraction one level higher by defining a `interceptReportForReactPlugin()` method?

Comment (francoismora) [reply eslint-bridge/src/linting/eslint/rules/decorators/no-unstable-nested-components.ts:31]: Indeed that helps clarify things by leaving the context overriding code only in the helper module. Thanks for the idea.

**Review by ilia-kebets-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**